src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSender.java

```diff
@@ -79,15 +86,28 @@ public void send(T value) throws IOException {
         requireNonNull(value, "Value for send can not be null");
 
         try {
-            ConnectionManager connection = this.connectionManager.get();
+            ConnectionManager connectionManager = this.connectionManager.get();
+            long messageCount = connectionManager.getMessageCount(queueName);
+            while (messageCount >= sizeLimit) {
```

Comment:
One routing key can be bound to several queues, for example, messages from the out_codec_decode pin are transferred to act and check1 as usual.

Reply:
There were 2 constructor params before my commit: `exchangeName` and `routingKey`.
I changed it to one `queueConfiguration` (also with `queueName`).

Reply:
The sender uses the basicPublish method to publish one message to RabbitMQ by routing key.
RabbitMQ puts a copy of the message to all queues to which the routing key is bound.

Reply:
done